app/javascript/admin/controllers/config_controller.js

```diff
@@ -119,7 +121,27 @@ export default class ConfigController extends Controller {
 
       displaySnackbar(outcome.message ?? outcome.error);
     } catch (err) {
+      errored = true;
       displaySnackbar(err.message);
+    } finally {
+      // Only update the site logo in the header if the new logo is uploaded successfully.
+      if (!errored) {
+        this.updateHeaderLogo(event.target);
+      }
+    }
+  }
+
+  updateHeaderLogo(form) {
+    if (form.elements.settings_general_logo) {
```

Comment:
Using the good old [HTMLFormElement.elements](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/elements) DOM API.